orchagent/aclorch.cpp

```diff
@@ -0,0 +1,1373 @@
+        errno = 0;
+        m_priority = strtol(attr_value.c_str(), &endp, 0);
+        // chack conversion was successfull and the value is within the allowed range
+        status = (errno == 0) &&
+                 (endp == attr_value.c_str() + attr_value.size()) &&
+                 (m_priority >= m_minPriority) &&
+                 (m_priority <= m_maxPriority);
+    }
+
+    return status;
+}
+
+bool AclRule::validateAddMatch(string attr_name, string attr_value)
+{
+    SWSS_LOG_ENTER();
+
+    sai_attribute_value_t value;
+
+    if (aclMatchLookup.find(attr_name) == aclMatchLookup.end())
+    {
+        return false;
+    }
+    else if(attr_name == MATCH_IP_TYPE)
+    {
+        if (!processIpType(attr_value, value.aclfield.data.u32))
+        {
+            SWSS_LOG_ERROR("Invalid IP type %s", attr_value.c_str());
+            return false;
+        }
+
+        value.aclfield.enable = true;
+        value.aclfield.mask.u32 = 0xFFFFFFFF;
+    }
+    else if(attr_name == MATCH_TCP_FLAGS)
+    {
+        vector<string> flagsData;
+        string flags, mask;
+        int val;
+        char *endp = NULL;
+        errno = 0;
+
+        split(attr_value, flagsData, '/');
+
+        if (flagsData.size() != 2) // expect two parts flags and mask separated with '/'
+        {
+            SWSS_LOG_ERROR("Invalid TCP flags format %s", attr_value.c_str());
+            return false;
+        }
+
+        flags = trim(flagsData[0]);
+        mask = trim(flagsData[1]);
+
+        val = strtol(flags.c_str(), &endp, 0);
+        if (errno || (endp != flags.c_str() + flags.size()) ||
+            (val < 0) || (val > UCHAR_MAX))
+        {
+            SWSS_LOG_ERROR("TCP flags parse error, value: %s(=%d), errno: %d", flags.c_str(), val, errno);
+            return false;
+        }
+        value.aclfield.data.u8 = val;
+
+        val = strtol(mask.c_str(), &endp, 0);
+        if (errno || (endp != mask.c_str() + mask.size()) ||
+            (val < 0) || (val > UCHAR_MAX))
+        {
+            SWSS_LOG_ERROR("TCP mask parse error, value: %s(=%d), errno: %d", mask.c_str(), val, errno);
+            return false;
+        }
+        value.aclfield.mask.u8 = val;
+    }
+    else if((attr_name == MATCH_ETHER_TYPE)  || (attr_name == MATCH_DSCP)        ||
+            (attr_name == MATCH_L4_SRC_PORT) || (attr_name == MATCH_L4_DST_PORT) ||
+            (attr_name == MATCH_IP_PROTOCOL))
+    {
+        char *endp = NULL;
+        errno = 0;
+        value.aclfield.data.u32 = strtol(attr_value.c_str(), &endp, 0);
+        if (errno || (endp != attr_value.c_str() + attr_value.size()))
+        {
+            SWSS_LOG_ERROR("Integer parse error. Attribute: %s, value: %s(=%d), errno: %d", attr_name.c_str(), attr_value.c_str(), value.aclfield.data.u32, errno);
+            return false;
+        }
+
+        value.aclfield.enable = true;
+        value.aclfield.mask.u32 = 0xFFFFFFFF;
+    }
+    else if (attr_name == MATCH_SRC_IP || attr_name == MATCH_DST_IP)
```

Comment:
Fixed.
Probably our SAI implementation ignores this flag so we did not catch this during testing